cpu/sam0_common/periph/uart.c

```diff
@@ -119,10 +119,10 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 void uart_write(uart_t uart, const uint8_t *data, size_t len)
 {
     for (size_t i = 0; i < len; i++) {
-        while (!(dev(uart)->INTFLAG.reg & SERCOM_USART_INTFLAG_DRE)) {}
+        while (!dev(uart)->INTFLAG.bit.DRE) {}
         dev(uart)->DATA.reg = data[i];
-        while (dev(uart)->INTFLAG.reg & SERCOM_USART_INTFLAG_TXC) {}
     }
+    while (!dev(uart)->INTFLAG.bit.TXC) {}
```

Comment:
nitpick: this now contains #7638...

Reply:
which is why I said above:

> Note: based on and includes #7638.

Reply:
> based on and includes #7638.

No it doesn't.

#7638 is broken and this PR is just re-doing the same change. If we'd merge #7638, you'd have to fix the one line it changes when rebasing this PR.

Please @smlng, there is a reason why we keep PR's small and don't piggyback style changes.

Reply:
okay factored fix out, now only replacing bit-fields here